spec/lib/logstash/bundler_spec.rb

```diff
@@ -27,4 +27,96 @@
       expect(exception.message).to eq("baz")
     end
   end
+  
+  context 'when invoking bundler' do
+    original_stderr = $stderr
+
+    subject { LogStash::Bundler.invoke_bundler!(options) }
+    # by default we want to fail fast on the test
+    let(:options) { { :install => true, :max_tries => 0, :without => [:development]} }
+    let(:bundler_args) { LogStash::Bundler.bundler_arguments(options) }
+
+    before do
+      $stderr = StringIO.new
+
+      expect(::Bundler).to receive(:reset!).at_least(1)
+    end
+
+    after do
+      expect(::Bundler.settings[:path]).to eq(LogStash::Environment::BUNDLE_DIR)
+      expect(::Bundler.settings[:gemfile]).to eq(LogStash::Environment::GEMFILE_PATH)
+      expect(::Bundler.settings[:without]).to eq(options.fetch(:without, []).join(':'))
+
+      expect(ENV['GEM_PATH']).to eq(LogStash::Environment.logstash_gem_home)
+
+      $stderr = original_stderr
+    end
+
+    it 'should call Bundler::CLI.start with the correct arguments' do
+      expect(::Bundler::CLI).to receive(:start).with(bundler_args)
+      subject
```

Comment:
Even if it's not wrong, somehow i read it wrong ... then there is another case when you call something like ``expect(subject)```